src/core/core_structs/simulations_stages.jl

```diff
@@ -1,24 +1,31 @@
 ######## Internal Simulation Object Structs ########
 abstract type AbstractStage end
 
-mutable struct _Stage <: AbstractStage
+mutable struct _Stage{M<:AbstractOperationModel} <: AbstractStage
     key::Int64
```

Comment:
Just a comment that you can take or leave:  stage_id or stage_index seems more descriptive for this variable.

Reply:
This is used in the code as stage.key. We would have then stage.stage_id.